qiime2/core/archive/archiver.py

```diff
@@ -462,11 +477,17 @@ def save(self, filepath):
         _ZipArchive.save(self.path, filepath)
 
     def validate_checksums(self):
-        if not isinstance(self._fmt, self.get_format_class('5')):
+        checksum_file = getattr(self._fmt, 'CHECKSUM_FILE', None)
+
+        if not checksum_file:
             return ChecksumDiff({}, {}, {})
 
-        obs = dict(x for x in md5sum_directory(str(self.root_dir)).items()
-                   if x[0] != self._fmt.CHECKSUM_FILE)
+        obs = \
+            dict(x for x in
+                 checksum_directory(str(self.root_dir),
+                                    checksum_type=self._fmt.CHECKSUM_TYPE)
+                 .items()
+                 if x[0] != self._fmt.CHECKSUM_FILE)
```

Comment:
We also need to exclude any annotations that may be present, otherwise it will raise an error here with "added" files.

Reply:
fixed from above comment re: annotations being excluded from checksum validation